### PR TITLE
Only give message about sleeping aid once

### LINKED ISSUE #37121: Redundant messages while sleeping
# Describe the bug

When I sleep with a pillow I get 12 messages per full hour of sleep saying "You use your pillow for comfort." Six dozen plus notifications about using a pillow for comfort can't be the intended behavior

# Steps To Reproduce

Sleep on a bed with a pillow. Happens every time for me

# Expected behavior

Ideally also describe *why* you expect it to happen.
I expect one message about using a pillow would be helpful to let me know that my character's sleep will be more restful.

# Screenshots
You use your pillow for comfort

# Versions and configuration

- OS: Windows 10
    - OS Version: 10.0 1903
- Game Version: 0.D-11310-g1fb22bc [64-bit]
- Graphics Version: Tiles
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Makeshift Items Mod [makeshift],
    Medieval and Historic Classes and Shields [Medieval_Stuff],
    More Survival Tools [More_Survival_Tools],
    Hydroponics [hydroponics],
    Mutant NPCs [mutant_npcs],
    More Locations [more_locations],
    Fuji's More Buildings [FujiStruct],
    More City Locations [cityside],
    Urban Development [Urban_Development],
    Folding Parts pack [deoxymod],
    Vehicle Additions Pack [blazemod],
    Tanks and Other Vehicles [Tanks],
    No Fungal Monsters [No_Fungi],
    Bionic Slots [cbm_slots],
    Classes and Scenarios Mod [more_classes_scenarios],
    Sleep Deprivation [sleepdeprivation],
    Stats Through Kills [stats_through_kills],
    Magiclysm [magiclysm],
    Modular Turrets [modular_turrets],
    Salvaged Robots [Salvaged_Robots],
    Roadheader and other mining vehicles [Heavy miners],
    SpeedyDex [speedydex],
    Amplified @'s Features [amplified_players_features],
    Stats Through Skills [StatsThroughSkills]
]

# Additional context

N/A

### PR DESCRIPTION
**Pull request by Fris0uman**
#### Summary
```SUMMARY: Bugfixes "Only give message about sleeping aid once"```

#### Purpose of change
Fixes #37121 

#### Describe the solution
If  `base_comfort_value` is called from update_needs don't print a message

#### Testing
Try to sleep on a pillow
Before fix get tons of message
After fix get only one message

### CODE REVIEW

**Comment by esotericist** [comment]
I'll give this a spin when I get back from errands.

good work on murdering more auto.

File: src/character.cpp [hunk 1]
```diff
@@ -4948,12 +4950,14 @@ Character::comfort_level Character::base_comfort_value( const tripoint &p ) cons
...
-        for( auto &items_it : items ) {
+        const map_stack items = g->m.i_at( p );
+        for( const item &items_it : items ) {
             if( items_it.has_flag( "SLEEP_AID" ) ) {
                 // Note: BED + SLEEP_AID = 9 pts, or 1 pt below very_comfortable
                 comfort += 1 + static_cast<int>( comfort_level::slightly_comfortable );
-                add_msg_if_player( m_info, _( "You use your %s for comfort." ), items_it.tname() );
+                if( message ) {
+                    add_msg_if_player( m_info, _( "You use your %s for comfort." ), items_it.tname() );
+                }
```
Comment (codemime) [inline src/character.cpp]: It would be nice to also deduplicate this snippet.

File: src/character.cpp [hunk 2]
```diff
@@ -4948,12 +4950,14 @@ Character::comfort_level Character::base_comfort_value( const tripoint &p ) cons
             comfort -= g->m.move_cost( p );
         }
 
-        auto items = g->m.i_at( p );
-        for( auto &items_it : items ) {
+        const map_stack items = g->m.i_at( p );
+        for( const item &items_it : items ) {
             if( items_it.has_flag( "SLEEP_AID" ) ) {
                 // Note: BED + SLEEP_AID = 9 pts, or 1 pt below very_comfortable
                 comfort += 1 + static_cast<int>( comfort_level::slightly_comfortable );
```
Comment (codemime) [inline src/character.cpp]: I can't test it right now, but I'm pretty sure there's a bug in here. One can put one `pillow` inside a vehicle and yet another under it, thus getting the bonus from both pillows (and two messages as a symptom).

File: src/character.cpp [hunk 3]
```diff
@@ -4878,14 +4878,16 @@ void Character::temp_equalizer( body_part bp1, body_part bp2 )
...
+Character::comfort_response_t Character::base_comfort_value( const tripoint &p ) const
 {
     // Comfort of sleeping spots is "objective", while sleep_spot( p ) is "subjective"
     // As in the latter also checks for fatigue and other variables while this function
     // only looks at the base comfyness of something. It's still subjective, in a sense,
     // as arachnids who sleep in webs will find most places comfortable for instance.
     int comfort = 0;
 
+    comfort_response_t comfort_response;
+    comfort_response.aid = &item( "null" );
```
Comment (BevapDin) [inline src/character.cpp]: This stores a pointer to a temporary object. This pointer becomes invalid right after this statement. I suggest you initialise it with a `nullptr` and check for that.

Comment (Fris0uman) [reply src/character.cpp]: Yep, it should be fixed now.

File: src/character.cpp [hunk 4]
```diff
@@ -4878,14 +4878,16 @@ void Character::temp_equalizer( body_part bp1, body_part bp2 )
...
-Character::comfort_level Character::base_comfort_value( const tripoint &p ) const
+Character::comfort_response_t Character::base_comfort_value( const tripoint &p ) const
 {
     // Comfort of sleeping spots is "objective", while sleep_spot( p ) is "subjective"
     // As in the latter also checks for fatigue and other variables while this function
     // only looks at the base comfyness of something. It's still subjective, in a sense,
     // as arachnids who sleep in webs will find most places comfortable for instance.
     int comfort = 0;
 
+    comfort_response_t comfort_response;
```
Comment (BevapDin) [inline src/character.cpp]: Now the pointer is uninitialised. Reading it is undefined behaviour.

**Pull request merged**